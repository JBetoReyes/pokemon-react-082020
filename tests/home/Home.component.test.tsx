import React from 'react';
import Home from '@app/home/Home.component';
import { IFetchState } from '@services/fetch.service';
import { IPokemonApp } from '@services/pokemon.service';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import useFetchPokemons from '@app/home/Home.hooks';

jest.mock('@app/home/Home.hooks');
const useFetchPokemonsMocked = useFetchPokemons as jest.MockedFunction<
  typeof useFetchPokemons
>;

describe('Home Component', () => {
  const mockState: IFetchState<IPokemonApp[]> = {
    state: 'loading',
    data: [],
  };
  useFetchPokemonsMocked.mockReturnValue(mockState);
  const homeJsx = <Home />;
  let home = shallow(homeJsx);
  beforeEach(() => {
    home = shallow(homeJsx);
    useFetchPokemonsMocked.mockClear();
  });
  test.skip('Should render the home component', () => {
    expect(home).toMatchSnapshot();
  });
  test.skip('Should render the home component with cards', () => {
    const mockStateWithPokemons: IFetchState<IPokemonApp[]> = {
      state: 'success',
      data: [
        {
          name: 'pikachu',
          number: 36,
          url: 'url',
        },
        {
          name: 'charmander',
          number: 37,
          url: 'url',
        },
      ],
    };
    useFetchPokemonsMocked.mockReturnValue(mockStateWithPokemons);
    home = shallow(homeJsx);
    expect(home.find('Card').length).toBe(2);
    expect(home).toMatchSnapshot();
  });
});
