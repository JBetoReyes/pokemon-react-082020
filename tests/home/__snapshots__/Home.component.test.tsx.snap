// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render the home component 1`] = `
<section>
  <SearchBar
    title="Which is your favorite pokemon?"
  />
  loading
</section>
`;

exports[`Home Component Should render the home component with cards 1`] = `
<section>
  <SearchBar
    title="Which is your favorite pokemon?"
  />
  <Carousel>
    <Card
      detail="pikachu"
      detailLabel="Name"
      key="36-pikachu"
      subDetail="36"
      subDetailLabel="Pokemon Number"
      url="https://pokeres.bastionbot.org/images/pokemon/36.png"
    />
    <Card
      detail="charmander"
      detailLabel="Name"
      key="37-charmander"
      subDetail="37"
      subDetailLabel="Pokemon Number"
      url="https://pokeres.bastionbot.org/images/pokemon/37.png"
    />
  </Carousel>
</section>
`;
